bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
Sorry, I meant that`get_cask_path` calls `get_cask_emacs` in this PR -- in the trunk of cask this knowledge is based implicitly through the environment. This fails with an --emacs command line.

Yes, I agree that the CASK variables are not altered once set. The EMACS variable would work in the same way. The only question is where the initialization happens. The obvious place being `main()`, yes.